docs/deprecations.md

```diff
@@ -25,6 +25,7 @@ The following features are deprecated but have not yet been removed.
 | [ClusterTask is deprecated](https://github.com/tektoncd/pipeline/issues/4476)                                                                                                                                                | v0.41.0                                                              | Beta                                                                                                   | July 13, 2023                       |
 | [`pipelineRef.bundle` and `taskRef.bundle` are deprecated](https://github.com/tektoncd/pipeline/issues/5514)                                                                                                                 | v0.41.0                                                              | Alpha                                                                                                  | July 13, 2023                       |
 | [The `config-trusted-resources` configMap is deprecated](https://github.com/tektoncd/pipeline/issues/5852)                                                                                                                 | v0.45.0                                                              | Alpha                                                                                                  | v0.46.0                       |
+| [v1beta1 Tasks, TaskRuns, Pipelines, and PipelineRuns are deprecated in favor of v1](https://github.com/tektoncd/pipeline/issues/5541)                                                                                                                 | v0.50.0                                                              | Beta                                                                                                  | v0.59.0                       |
```

Comment:
Is this accurate? The deprecation is announced in 0.50 which is LTS and will have the estimated end of the life in a year around `2024-07-20` (might be v0.62.0) unless we cut release v1 before that.

Reply:
Currently our policy for beta CRDs is 9 months notice, or 9 releases. So the last release with support would be 0.59.0, which would have an end of life of 1 yr + 9 months from the date 0.50.0 is released. I think having the last release with support listed here makes sense-- if we put a date I'm worried people might think we cannot remove the CRDs until that date.

Reply:
I am not sure if I am following, 1 year + 9 months from the date 0.50 is released. 

Yup our policy for beta CRDs is 9 months/releases.

Let's take walk through this scenario. There is a bug discovered in 0.50 with `v1beta1` by the time we are about to release 0.60 (May, 2024). Do we keep `v1beta1` CRDs around in `0.60` to fix such potential bugs or those CRDs are deleted in `0.59`? The way we are cherry picking after fixing an issue in `main`, are we going to follow the same process? If we are going to follow the same process and if we delete `v1beta1` CRDs, how can we fix a bug in `0.50`? (lot of ifs 🤣) Does it make sense? Am I complicating it?

Reply:
> There is a bug discovered in 0.50 with v1beta1 by the time we are about to release 0.60 (May, 2024). Do we keep v1beta1 CRDs around in 0.60 to fix such potential bugs or those CRDs are deleted in 0.59?

If we want to backport a fix to 0.50, we could just create a commit on the release-v0.50.x branch. I don't think we need to keep the CRDs around on the main branch in case we need to backport a bug fix.

I'm curious if there's confusion here due to the client libraries + go structs, vs the actual CRD and whether it's served? In https://github.com/tektoncd/pipeline/pull/6768, I updated our policy to say that we'd keep the Go clients for v1beta1 CRDs until the last release when they are served (v0.59.0) reaches its EOL (1 year later, or when we are releasing v0.69.0).